test/src/main/java/com/coderanch/test/AbstractComparableTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.test;
+
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theory;
+
+import static java.lang.Integer.signum;
+
+import static org.hamcrest.Matchers.comparesEqualTo;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeNoException;
+import static org.junit.Assume.assumeNotNull;
+import static org.junit.Assume.assumeThat;
+
+@SuppressWarnings({ "rawtypes", "unchecked", "squid:S00100"})
+public abstract class AbstractComparableTest extends AbstractObjectTest {
+
+    @Theory
+    public final void compareTo_isReflexive(
+        @FromDataPoints("objects") Comparable x
+    ) {
+        assumeNotNull(x);
+        assertThat(x, comparesEqualTo(x));
+    }
+
+    @Theory
+    public final void compareTo_isSymmetric(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Document this public method by adding an explicit description. ([squid:UndocumentedApi](https://sonarcloud.io/organizations/coderanch-corral/rules#rule_key=squid%3AUndocumentedApi))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.coderanch%3Ablackjack-parent&pullRequest=11&issues=AWaB525_ksgJibV5mvFO&open=AWaB525_ksgJibV5mvFO)

[ISSUE_KEY]: AWaB525_ksgJibV5mvFO